source/read/cursors.txt

```diff
@@ -0,0 +1,171 @@
+.. _kotlin-sync-cursors:
+
+=========================
+Access Data From a Cursor
+=========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: read, results, oplog
+
+Overview
+--------
+
+In this guide, you can learn how to access data from a **cursor** with the
+{+driver-short+}.
+
+A cursor is a mechanism that returns the results of a read operation in iterable
+batches. Because a cursor holds only a subset of documents at any given time,
+cursors reduce both memory consumption and network bandwidth usage.
+
+Whenever the {+driver-short+} performs a read operation that returns multiple
+documents, it automatically returns those documents in a cursor.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+The following {+language+} data class models the documents in this collection:
+
+.. literalinclude:: /includes/read/cursors.kt
+   :start-after: start-data-class
+   :end-before: end-data-class
+   :language: kotlin
+   :copyable:
+
+.. _kotlin-sync-cursors-iterate:
+
+Access Cursor Contents Iteratively
+----------------------------------
+
+To iterate over the contents of a cursor, use the ``forEach()`` method, as shown in the
+following example:
+
+.. literalinclude:: /includes/read/cursors.kt
+   :start-after: start-cursor-iterate
+   :end-before: end-cursor-iterate
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+Retrieve Documents Individually
```

Comment:
Calling `hasNext`/`next` in a loop (it is supposed to be called in a loop) is just an alternative way of iterating the cursor. It seems wrong to imply that only `forEach` is the iterative approach.

Reply:
I can provide an alternate example using a `while` loop with `hasNext` and `next`.